src/main/java/com/minecolonies/coremod/colony/TriggerColonyAchievements.java

```diff
@@ -0,0 +1,246 @@
+
+/**
+ * Trigger the corresponding colony achievement.
+ */
+public final class TriggerColonyAchievements
+{
+    private static final String TAG_LUMBERJACK_SAPLINGS   = "saplings";
+    private static final String TAG_LUMBERJACK_TREES      = "trees";
+    private static final String TAG_FISHERMAN_FISH        = "fish";
+    private static final String TAG_BUILDER_HUTS          = "huts";
+    private static final String TAG_GUARD_MOBS            = "mobs";
+    private static final String TAG_FARMER_CARROTS        = "carrots";
+    private static final String TAG_FARMER_POTATOES       = "potatoes";
+    private static final String TAG_FARMER_WHEAT          = "wheat";
+    private static final String TAG_MINER_DIAMONDS        = "diamonds";
+    private static final String TAG_MINER_ORES            = "ores";
+
+    private TriggerColonyAchievements()
+    {
+        throw new IllegalAccessError("Utility class");
+    }
+
+    /**
+     *Trigger fifth achievement.
+     * @param statistic the statistic.
+     * @param colony the colony.
+     */
+    public static void triggerFirstAchievement(@NotNull String statistic, @NotNull Colony colony)
+    {
+        switch (statistic)
+        {
+            case TAG_GUARD_MOBS:
+                colony.triggerAchievement(ModAchievements.achievementKillOneMob);
+                break;
+            case TAG_MINER_ORES:
+                colony.triggerAchievement(ModAchievements.achievementMineOneOre);
+                break;
+            case TAG_MINER_DIAMONDS:
+                colony.triggerAchievement(ModAchievements.achievementMineOneDiamond);
+                break;
+            case TAG_BUILDER_HUTS:
+                colony.triggerAchievement(ModAchievements.achievementBuildOneHut);
+                break;
+            case TAG_FISHERMAN_FISH:
+                colony.triggerAchievement(ModAchievements.achievementCatchOneFish);
+                break;
+            case TAG_FARMER_WHEAT:
+                colony.triggerAchievement(ModAchievements.achievementHarvestOneWheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                colony.triggerAchievement(ModAchievements.achievementHarvestOnePotato);
+                break;
+            case TAG_FARMER_CARROTS:
+                colony.triggerAchievement(ModAchievements.achievementHarvestOneCarrot);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                colony.triggerAchievement(ModAchievements.achievementPlantOneSapling);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                colony.triggerAchievement(ModAchievements.achievementFellOneTree);
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     *Trigger fifth achievement.
+     * @param statistic the statistic.
+     * @param colony the colony.
+     */
+    public static void triggerSecondAchievement(@NotNull String statistic, @NotNull Colony colony)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerSecondAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)